pyingest/parsers/aps.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+
+import sys
+import json
+import codecs
+from adsputils import u2asc
+from jats import JATSParser
+
+class NoSchemaException(Exception):
+    pass
+
+class WrongSchemaException(Exception):
+    pass
+
+class UnparseableException(Exception):
+    pass
+
+
+class APSJATSParser(JATSParser):
+
+    def get_author_init(self,namestring):
+        output = u2asc(namestring)
+        for c in output:
+            if c.isalpha():
+                return c
+        return u'.'
+
+    def aps_journals(self, pid):
+#       mapping journal-meta/journal-id/publisher-id to bibstems
+        publisher_ids = {'PRL': 'PhRvL', 'PRX': 'PhRvX', 'RMP': 'RvMP',
+                         'PRA': 'PhRvA', 'PRB': 'PhRvB', 'PRC': 'PhRvC',
+                         'PRD': 'PhRvD', 'PRE': 'PhRvE', 'PRAB': 'PhRvS',
+                         'PRSTAB': 'PhRvS', 'PRAPPLIED': 'PhRvP',
+                         'PRFLUIDS': 'PhRvF', 'PRMATERIALS': 'PhRvM', 
+                         'PRPER': 'PRPER', 'PRSTPER': 'PRSTP', 'PR': 'PhRv',
+                         'PRI': 'PhRvI','PHYSICS': 'PhyOJ'}
+        try:
+            bibstem = publisher_ids[pid]
+        except KeyError:
+            return 'XSTEM'
+        else:
+            return bibstem
+
+#   def make_journal_field
+
+    def doi_parse(self, doi):
```

Comment:
I don't understand why we are extracting volume and id from the DOI, given that they are found in the source metadata already.  Given that DOIs are meant to be opaque, there is no guarantee that relying on the current recipe is robust going forward.